### PR TITLE
Fix locations of block quotes

### PR DESCRIPTION
**Pull request by panglesd**
Previously, they did not include the initial marker:

In:

```markdown
> Hello
>
> Goodbye
```

the location of the block quote started at "`H`" instead of the first "`>`".

### CODE REVIEW

**Comment by dbuenzli** [comment]
I can't recall exactly why but to me it looks like the current behaviour was made on purpose. The text location wraps the quoted block. If you want to get to the marker you can substract [`Block_quote.indent`](https://erratique.ch/software/cmarkit/doc/Cmarkit/Block/Block_quote/index.html#val-indent)`+ 1`

**Comment by dbuenzli** [comment]
Ah no sorry `Block_quote.indent` is the indent to the marker.

**Comment by panglesd** [comment]
Yes! I think that in the current state, the location of the block quote is exactly the location of the block in it.

(This is the case in all the tests here, but there might be some corner case I'm missing where it's not the case?)

File: src/cmarkit.ml [hunk 1]
```diff
@@ -2056,7 +2060,7 @@ module Block_struct = struct
   type paragraph = { maybe_ref : bool; lines : line_span list }
 
   type t =
-  | Block_quote of Layout.indent * t list
+  | Block_quote of Layout.indent * t list * line_span (* loc of initial marker *)
```
Comment (dbuenzli) [inline src/cmarkit.ml]: Could you please put the `line_span` after the indent, so that it reads in order.

File: test/snapshots/basic.nolayout.locs [hunk 2]
```diff
@@ -493,9 +493,9 @@ File "basic.md", lines 1-195
   Blank line:
   File "basic.md", line 84
   Block quote:
-  File "basic.md", lines 85-88, characters 8-43
+  File "basic.md", lines 85-88, characters 2-43
     Block quote:
-    File "basic.md", lines 85-88, characters 8-43
+    File "basic.md", lines 85-88, characters 6-43
```
Comment (dbuenzli) [inline test/snapshots/basic.nolayout.locs:498]: This looks like a nice improvement for nested blocks on the first line. Before everything would collapse to the start of the innermost block.

**Review by dbuenzli — CHANGES_REQUESTED** [review]
I think I'm convinced by your changes I just have a (rather annoying :-) cosmetic comment.

**Comment by dbuenzli** [comment]
> (This is the case in all the tests here, but there might be some corner case I'm missing where it's not the case?)

I don't think so. I'm not sure exactly why I didn't do what you suggest. I checked with other constructs (e.g. headings/item lists) and usually the textloc spans over the structural element (`#`, item marker), except for indented code blocks.

File: src/cmarkit.ml (hunk 1 above)
Comment (panglesd) [reply src/cmarkit.ml]: Done in 178f2cc

File: src/cmarkit.ml [hunk 3]
```diff
@@ -1980,13 +1980,17 @@ module Block_struct = struct
 
   let match_and_accept_block_quote p =
     (* https://spec.commonmark.org/current/#block-quote-marker *)
-    if end_of_line p || p.i.[p.current_char] <> '>' then false else
+    if end_of_line p || p.i.[p.current_char] <> '>' then Match.Nomatch else
+    let marker_span =
+      current_line_span p ~first:p.current_char ~last:(p.current_char+1)
```
Comment (dbuenzli) [inline src/cmarkit.ml]: `last` should be `p.current_char` but I can correct that on merge.

Comment (panglesd) [reply src/cmarkit.ml]: Thanks for pointing it out! (since I need to fix it in my fork too)

I pushed c2b1625 with the fix if that helps.

**Comment by dbuenzli** [comment]
Thanks your patch is in as  2dd8dbab4ced
